src/pages/addProduct/index.js

```diff
@@ -0,0 +1,95 @@
+import { useRef, useState } from "react";
+import { useHistory } from "react-router";
+import { useEffect } from "react/cjs/react.development";
+import { ErrorField } from "../../components/elements";
+import { HOMEPAGE } from "../../constants";
+import { addProduct } from "../../services/productService";
+import { isContentManager } from "../../utils";
+import "./addProduct.css";
+
+export function AddProduct() {
+  const history = useHistory();
+  const [nameError, setNameError] = useState("");
+  const formRef = useRef();
+  useEffect(() => {
+    if (!isContentManager()) {
+      history.replace(HOMEPAGE);
+    }
+  });
```

Comment:
When you don't pass dependency array, it means callback will be executed on each render which is not intentional behavior, pass an empty array as a dependency array to execute callback only on the first render. Leave blank lines before and after this code block.
```suggestion
  useEffect(() => {
    if (!isContentManager()) {
      history.replace(HOMEPAGE);
    }
  }, []);
```